gc/base/MemorySubSpaceGenerational.cpp

```diff
@@ -323,9 +323,16 @@ MM_MemorySubSpaceGenerational::counterBalanceContract(
 }
 
 uintptr_t
-MM_MemorySubSpaceGenerational::releaseFreeMemoryPages(MM_EnvironmentBase* env)
+MM_MemorySubSpaceGenerational::releaseFreeMemoryPages(MM_EnvironmentBase* env, uintptr_t memoryType)
 {
-	return _memorySubSpaceOld->releaseFreeMemoryPages(env);
+	uintptr_t releasedPages = 0;
+	if (OMR_ARE_ALL_BITS_SET(memoryType, MEMORY_TYPE_OLD)) {
+		releasedPages += _memorySubSpaceOld->releaseFreeMemoryPages(env);
```

Comment:
seems like we could get rid of plain variant of this API for Flat if we call it here the variant with parameter (and specify OLD). no strong preference - up to you.